tests/test_ppo_trainer.py

```diff
@@ -1151,7 +1150,6 @@ def test_grad_accumulation(self):
             reward = [torch.tensor(1.0), torch.tensor(1.0)]
             # train model by running a step twice
             _ = ppo_trainer.step([q for q in query_tensor], [r for r in response_tensor], reward)
-            break
 
         model_grad_acc = gpt2_model_clone.v_head.summary.weight.grad.clone()
         self.assertTrue(torch.allclose(model_grad_acc, model_grad, rtol=1e-3, atol=1e-3))
```

Comment:
A quick comment here: at some point I tried printing out the grads and they are all zeros here — I think it’s because we already did `optimizer.step`. We should be comparing the weights instead: 

```
self.assertTrue(torch.allclose(gpt2_model_clone.v_head.summary.weight, gpt2_model.v_head.summary.weight, rtol=1e-3, atol=1e-3))
```

Reply:
Thanks, added that suggestion !